File: problems/6/6.py
# coding: utf-8

"""
The sum of the squares of the first ten natural numbers is,

12 + 22 + ... + 102 = 385

The square of the sum of the first ten natural numbers is,

(1 + 2 + ... + 10)2 = 552 = 3025

Hence the difference between the sum of the squares of the first ten natural
numbers and the square of the sum is 3025 − 385 = 2640.

Find the difference between the sum of the squares of the first one hundred
natural numbers and the square of the sum.
"""

import time


def oneliner():
    return sum([x**2 for x in range(100+1)]) - sum(range(100+1))**2


def solve():
    def sum_of_squares(n):
        squares = []
        for i in range(n+1):
            squares.append(i**2)
        return sum(squares)

    def square_of_sum(n):
        return sum(range(n+1))**2

    return sum_of_squares(100) - square_of_sum(100)


if __name__ == "__main__":
    def timeit(function):
        t1 = time.time()
        output = function()
        t2 = time.time()
        return output, t2-t1

    print timeit(solve)
    print timeit(oneliner)

File: problems/6/6.js
//     node 6.js

// Problem:
//
// The sum of the squares of the first ten natural numbers is,
//
// 1² + 2² + ... + 10² = 385
//
// The square of the sum of the first ten natural numbers is,
//
// (1 + 2 + ... + 10)² = 552 = 3025
//
// Hence the difference between the sum of the squares of the first ten natural
// numbers and the square of the sum is 3025 − 385 = 2640.
//
// Find the difference between the sum of the squares of the first one hundred
// natural numbers and the square of the sum.

solve = function() {
    var sum = 0,
        sum_of_squares = 0,
        N = 100,
        result;

    for(i=0; i<=N; i++){
        sum += i;
        sum_of_squares += i*i;
    }

    result = sum_of_squares - sum*sum;
    return Math.abs(result)
}



// unfortunately the accurracy seems to be only 1ms.
console.time('solver');
console.log(solve());
console.timeEnd('solver');
